torch/_export/serde/schema.py

```diff
@@ -154,15 +148,23 @@ class GraphSignature:
     user_inputs: List[str]
     user_outputs: List[str]
     buffers_to_mutate: Dict[str, str]
+    backward_signature: Optional[BackwardSignature]
+
+
+@dataclass
+class CallSpec:
     in_spec: str
     out_spec: str
-    backward_signature: Optional[BackwardSignature]
 
 
 @dataclass
 class GraphModule:
     graph: Graph
-    buffers: Dict[str, TensorMeta]
-    parameters: Dict[str, TensorMeta]
-    metadata: Dict[str, str]
+    opset_version: Dict[str, int]
+
+
+@dataclass
+class ExportedProgram:
+    graph_module: GraphModule
     signature: GraphSignature
+    call_spec: CallSpec
```

Comment:
Can you describe what the purpose of `call_spec` is? Won't `graph_module` have the `call_spec` mappings burned in?

Is there a standard transform that uses this information?

Reply:
call_spec contains the in_spec/out_spec which is used when we're running the graph module eagerly so that the inputs/output formats match how we would run the original function eagerly.

Reply:
Bit confused about why we also then have flatten / unflatten calls in a graph module...

Reply:
Mm, it defines the way we pass inputs and receive outputs to the graph module. Zhengxu stated below that it makes more sense to put the GraphSignature in GraphModule, and I think CallSpec should go with wherever the Signature goes.